packages/navigation/src/buildWebScreen.tsx

```diff
@@ -1,28 +1,28 @@
 import {
-  PartialRoute,
-  Route,
   getStateFromPath,
+  PartialState,
 } from '@react-navigation/native';
+import type { NavigationState } from '@react-navigation/core';
 
 import { LinkingConfig } from './hooks/useCurrentUrl';
 import { unpackState } from './utils/unpackState';
 
 type Options = Parameters<typeof getStateFromPath>[1];
 type LinkedParams = { baseURL?: string; fullPath?: string };
-type PartialRoutes = readonly PartialRoute<Route<string, object | undefined>>[];
 
 function getParams(
-  routes: PartialRoutes | undefined
+  state: NavigationState | PartialState<NavigationState>
 ): LinkedParams | undefined {
-  const firstRoute = routes?.[0];
-  if (firstRoute) {
-    if (firstRoute.params) {
-      return firstRoute.params;
+  const activeRoute = state.routes?.[state.index ?? 0];
```

Comment:
This is the prominent change; previously we always returned the params for the _first_ route, not the active route. This PR changes so we use `state.index` to determine the active route.